### ISSUE TITLE
Move to azure vm scale sets

### ISSUE THREAD

**Issue by Mpdreamz** [elastic/azure-marketplace#30]
https://azure.microsoft.com/en-us/blog/azure-vm-scale-sets-public-preview/

Does not support premium storage just yet but would greatly simplify and cut back on the moving parts in the ARM template

**Comment by niemyjski** [comment]
I wonder if they implemented this yet, I tried to do some searches but didn't really seeing anything explicitly stating this.

**Comment by ajhewett** [comment]
VMSS with Managed Disks went [GA in Feb 2017](https://azure.microsoft.com/en-us/blog/announcing-general-availability-of-managed-disks-and-larger-scale-sets/).
Is anybody working on this? If not, I may be able to contribute something.

**Comment by russcam** [comment]
@ajhewett We're looking into implementing VM Scale sets for data nodes

**Comment by makhdumi** [comment]
Just curious - I'm assuming that this will only work with scaling out, correct?

I've been scouring the documentation and cannot find any way to run something when a node is about deprovisioned in a Scale Set, nor any way to specify a preference for which nodes should be deprovisioned.

**Comment by darrell-tethr** [comment]
Can someone please expound on this?  We are planning to use Scale Sets for Client and Ingest Nodes in our ES cluster (and possibly Data nodes in the future as well).

**Comment by russcam** [comment]
@makhdumi correct, scaling up should be fine with VM scale sets, but scaling down data nodes will need to be investigated. Recently, the fault and update domains for the VM have been included within the Elasticsearch node attributes to configure shard allocation awareness, but it remains to be investigated what other levels of control VM scale sets can provide to deprovision specific nodes in a scale down scenario.

@darrell-tethr client (coordinating) nodes and ingest only nodes _should_ work for both scale up and down with VM scale sets, since neither hold index data.

**Comment by mikepetridisz** [comment]
I see it's still an open topic. I know managed elastic on Azure is rolled out. However, for us, only Elasticsearch (Self-Managed) is a viable option due to compliance reasons. Is there any update or plan around VMSS?
